// @prisma/estoque-client/wasm.js

/* !!! This is code generated by Prisma. Do not edit directly. !!!
/* eslint-disable */

Object.defineProperty(exports, "__esModule", { value: true });

const {
  PrismaClientKnownRequestError,
  PrismaClientUnknownRequestError,
  PrismaClientRustPanicError,
  PrismaClientInitializationError,
  PrismaClientValidationError,
  getPrismaClient,
  sqltag,
  empty,
  join,
  raw,
  skip,
  Decimal,
  Debug,
  objectEnumValues,
  makeStrictEnum,
  Extensions,
  warnOnce,
  defineDmmfProperty,
  Public,
  getRuntime,
  createParam,
} = require('./runtime/wasm-engine-edge.js')


const Prisma = {}

exports.Prisma = Prisma
exports.$Enums = {}

/**
 * Prisma Client JS version: 6.17.0
 * Query Engine version: c0aafc03b8ef6cdced8654b9a817999e02457d6a
 */
Prisma.prismaVersion = {
  client: "6.17.0",
  engine: "c0aafc03b8ef6cdced8654b9a817999e02457d6a"
}

Prisma.PrismaClientKnownRequestError = PrismaClientKnownRequestError;
Prisma.PrismaClientUnknownRequestError = PrismaClientUnknownRequestError
Prisma.PrismaClientRustPanicError = PrismaClientRustPanicError
Prisma.PrismaClientInitializationError = PrismaClientInitializationError
Prisma.PrismaClientValidationError = PrismaClientValidationError
Prisma.Decimal = Decimal

/**
 * Re-export of sql-template-tag
 */
Prisma.sql = sqltag
Prisma.empty = empty
Prisma.join = join
Prisma.raw = raw
Prisma.validator = Public.validator

/**
* Extensions
*/
Prisma.getExtensionContext = Extensions.getExtensionContext
Prisma.defineExtension = Extensions.defineExtension

/**
 * Shorthand utilities for JSON filtering
 */
Prisma.DbNull = objectEnumValues.instances.DbNull
Prisma.JsonNull = objectEnumValues.instances.JsonNull
Prisma.AnyNull = objectEnumValues.instances.AnyNull

Prisma.NullTypes = {
  DbNull: objectEnumValues.classes.DbNull,
  JsonNull: objectEnumValues.classes.JsonNull,
  AnyNull: objectEnumValues.classes.AnyNull
}





/**
 * Enums
 */
exports.Prisma.TransactionIsolationLevel = makeStrictEnum({
  Serializable: 'Serializable'
});

exports.Prisma.ProdutoScalarFieldEnum = {
  id: 'id',
  nome: 'nome',
  marca: 'marca',
  categoria: 'categoria',
  unidade: 'unidade',
  quantidadeMin: 'quantidadeMin',
  fornecedor: 'fornecedor',
  observacoes: 'observacoes',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.EntradaScalarFieldEnum = {
  id: 'id',
  produtoId: 'produtoId',
  quantidade: 'quantidade',
  data: 'data',
  fornecedor: 'fornecedor',
  precoPago: 'precoPago',
  responsavelId: 'responsavelId'
};

exports.Prisma.SaidaScalarFieldEnum = {
  id: 'id',
  produtoId: 'produtoId',
  quantidade: 'quantidade',
  data: 'data',
  responsavelId: 'responsavelId',
  motivo: 'motivo'
};

exports.Prisma.HistoricoPrecoScalarFieldEnum = {
  id: 'id',
  produtoId: 'produtoId',
  preco: 'preco',
  data: 'data'
};

exports.Prisma.SortOrder = {
  asc: 'asc',
  desc: 'desc'
};

exports.Prisma.NullsOrder = {
  first: 'first',
  last: 'last'
};


exports.Prisma.ModelName = {
  Produto: 'Produto',
  Entrada: 'Entrada',
  Saida: 'Saida',
  HistoricoPreco: 'HistoricoPreco'
};
/**
 * Create the Client
 */
const config = {
  "generator": {
    "name": "client",
    "provider": {
      "fromEnvVar": null,
      "value": "prisma-client-js"
    },
    "output": {
      "value": "/workspaces/codespaces-blank/backend/node_modules/@prisma/estoque-client",
      "fromEnvVar": null
    },
    "config": {
      "engineType": "library"
    },
    "binaryTargets": [
      {
        "fromEnvVar": null,
        "value": "debian-openssl-1.1.x",
        "native": true
      }
    ],
    "previewFeatures": [],
    "sourceFilePath": "/workspaces/codespaces-blank/backend/prisma/estoque/schema.prisma",
    "isCustomOutput": true
  },
  "relativeEnvPaths": {
    "rootEnvPath": "../../../.env",
    "schemaEnvPath": "../../../.env"
  },
  "relativePath": "../../../prisma/estoque",
  "clientVersion": "6.17.0",
  "engineVersion": "c0aafc03b8ef6cdced8654b9a817999e02457d6a",
  "datasourceNames": [
    "db"
  ],
  "activeProvider": "sqlite",
  "postinstall": false,
  "inlineDatasources": {
    "db": {
      "url": {
        "fromEnvVar": null,
        "value": "file:./estoque.db"
      }
    }
  },
  "inlineSchema": "generator client {\n  provider = \"prisma-client-js\"\n  // gera o client em uma pasta diferente para evitar conflito\n  output   = \"../../node_modules/@prisma/estoque-client\"\n}\n\ndatasource db {\n  provider = \"sqlite\"\n  url      = \"file:./estoque.db\"\n}\n\nmodel Produto {\n  id            Int     @id @default(autoincrement())\n  nome          String\n  marca         String?\n  categoria     String?\n  unidade       String // ex: \"kg\", \"L\", \"unid\"\n  quantidadeMin Int\n  fornecedor    String?\n  observacoes   String?\n\n  entradas       Entrada[]\n  saidas         Saida[]\n  historicoPreco HistoricoPreco[]\n\n  createdAt DateTime @default(now())\n  updatedAt DateTime @updatedAt\n}\n\nmodel Entrada {\n  id            Int      @id @default(autoincrement())\n  produtoId     Int\n  produto       Produto  @relation(fields: [produtoId], references: [id])\n  quantidade    Int\n  data          DateTime @default(now())\n  fornecedor    String?\n  precoPago     Float\n  responsavelId Int // referência ao id de usuário (vindo do outro banco, manual)\n}\n\nmodel Saida {\n  id            Int      @id @default(autoincrement())\n  produtoId     Int\n  produto       Produto  @relation(fields: [produtoId], references: [id])\n  quantidade    Int\n  data          DateTime @default(now())\n  responsavelId Int // referência ao id de usuário (vindo do outro banco)\n  motivo        String?\n}\n\nmodel HistoricoPreco {\n  id        Int      @id @default(autoincrement())\n  produtoId Int\n  produto   Produto  @relation(fields: [produtoId], references: [id])\n  preco     Float\n  data      DateTime @default(now())\n}\n",
  "inlineSchemaHash": "ad3cbb7232fb127b3dd55e09d1e0e97ca9755fc62c4cf7e197de393b85324d63",
  "copyEngine": true
}
config.dirname = '/'

config.runtimeDataModel = JSON.parse("{\"models\":{\"Produto\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"nome\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"marca\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"categoria\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"unidade\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"quantidadeMin\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"fornecedor\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"observacoes\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"entradas\",\"kind\":\"object\",\"type\":\"Entrada\",\"relationName\":\"EntradaToProduto\"},{\"name\":\"saidas\",\"kind\":\"object\",\"type\":\"Saida\",\"relationName\":\"ProdutoToSaida\"},{\"name\":\"historicoPreco\",\"kind\":\"object\",\"type\":\"HistoricoPreco\",\"relationName\":\"HistoricoPrecoToProduto\"},{\"name\":\"createdAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"updatedAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"}],\"dbName\":null},\"Entrada\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"produtoId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"produto\",\"kind\":\"object\",\"type\":\"Produto\",\"relationName\":\"EntradaToProduto\"},{\"name\":\"quantidade\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"data\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"fornecedor\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"precoPago\",\"kind\":\"scalar\",\"type\":\"Float\"},{\"name\":\"responsavelId\",\"kind\":\"scalar\",\"type\":\"Int\"}],\"dbName\":null},\"Saida\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"produtoId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"produto\",\"kind\":\"object\",\"type\":\"Produto\",\"relationName\":\"ProdutoToSaida\"},{\"name\":\"quantidade\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"data\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"responsavelId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"motivo\",\"kind\":\"scalar\",\"type\":\"String\"}],\"dbName\":null},\"HistoricoPreco\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"produtoId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"produto\",\"kind\":\"object\",\"type\":\"Produto\",\"relationName\":\"HistoricoPrecoToProduto\"},{\"name\":\"preco\",\"kind\":\"scalar\",\"type\":\"Float\"},{\"name\":\"data\",\"kind\":\"scalar\",\"type\":\"DateTime\"}],\"dbName\":null}},\"enums\":{},\"types\":{}}")
defineDmmfProperty(exports.Prisma, config.runtimeDataModel)
config.engineWasm = {
  getRuntime: async () => require('./query_engine_bg.js'),
  getQueryEngineWasmModule: async () => {
    const loader = (await import('#wasm-engine-loader')).default
    const engine = (await loader).default
    return engine
  }
}
config.compilerWasm = undefined

config.injectableEdgeEnv = () => ({
  parsed: {}
})

if (typeof globalThis !== 'undefined' && globalThis['DEBUG'] || typeof process !== 'undefined' && process.env && process.env.DEBUG || undefined) {
  Debug.enable(typeof globalThis !== 'undefined' && globalThis['DEBUG'] || typeof process !== 'undefined' && process.env && process.env.DEBUG || undefined)
}

const PrismaClient = getPrismaClient(config)
exports.PrismaClient = PrismaClient
Object.assign(exports, Prisma)
